ofrak_core/ofrak/gui/server.py

```diff
@@ -739,6 +742,61 @@ async def search_for_vaddr(self, request: Request) -> Response:
         except NotFoundError:
             return json_response([])
 
+    @exceptions_to_http(SerializedError)
+    async def search_for_string(self, request: Request):
+        resource = await self._get_resource_for_request(request)
+        body = await request.json()
+        string_query_string = body["search_query"]
+        regex = body["regex"]
+        case_ignore = body["case_ignore"]
+        if not isinstance(string_query_string, str):
+            raise ValueError("Invalid search query.")
+        string_query: Union[bytes, re.Pattern[bytes]] = string_query_string.encode()
+        if case_ignore:
+            string_query = re.compile(string_query, re.IGNORECASE)
+        elif regex:
+            string_query = re.compile(string_query)
+
+        offsets = await resource.search_data(string_query)
+        if string_query == "":
+            return json_response(None)
```

Comment:
Is there a reason to return `None` instead of an empty list here? 

Also, if `string_query == ""`, I think it makes sense to return earlier than this. Not only does it make sense to return before actually doing the search (to avoid the expensive operation of searching for an empty string that will be present between every single character), but it probably makes sense to return before compiling the regular expression as well.

Reply:
On the first part, yes. An empty list should be the result of a search with no results. Otherwise it isn't apparent that your search is empty. And yeah you're correct, I did this fast this morning.

Reply:
Second part sounds good, makes sense to return `None`.